nodelet/pkg/phases/container_runtime/start_containerd.go

```diff
@@ -0,0 +1,146 @@
+	}
+	return runtimeStartPhase
+}
+
+func (cp *ContainerdRunPhase) GetHostPhase() sunpikev1alpha1.HostPhase {
+	return *cp.hostPhase
+}
+
+func (cp *ContainerdRunPhase) GetPhaseName() string {
+	return cp.hostPhase.Name
+}
+
+func (cp *ContainerdRunPhase) GetOrder() int {
+	return int(cp.hostPhase.Order)
+}
+
+func (cp *ContainerdRunPhase) Start(ctx context.Context, cfg config.Config) error {
+
+	cp.log.Infof("Running Start of phase: %s", cp.hostPhase.Name)
+	//TODO: configure_containerd_http_proxy
+
+	cp.log.Infof("Starting containerd")
+	exitCode, err := cp.cmd.RunCommand(ctx, nil, 0, "", "/bin/sudo", "/usr/bin/systemctl", "start", constants.RuntimeContainerd)
+	if exitCode != 0 || err != nil {
+		cp.log.Errorf("error starting containerd:%v comand exited with exitcode:%v", err, exitCode)
+		phaseutils.SetHostStatus(cp.hostPhase, constants.FailedState, fmt.Sprintf("couldn't start containerd:%v", err))
+		return errors.Wrapf(err, "error starting containerd command exited with exitcode:%v", exitCode)
+
+	}
+	cp.log.Infof("Started containerd")
+
+	//TODO: login to Dockerhub(if necessary)
+	phaseutils.SetHostStatus(cp.hostPhase, constants.RunningState, "")
+	return nil
+}
+
+func (cp *ContainerdRunPhase) Stop(ctx context.Context, cfg config.Config) error {
+
+	cp.log.Infof("Running Stop of phase: %s", cp.hostPhase.Name)
+
+	exitCode, _, err := cp.cmd.RunCommandWithStdOut(ctx, nil, 0, "", constants.RuntimeContainerd, "--version")
+
+	if err != nil || exitCode != 0 {
+		cp.log.Warn("containerd not present so cant destroy containers: %v", err)
+		phaseutils.SetHostStatus(cp.hostPhase, constants.StoppedState, "")
+		return nil
+	}
+
+	containerUtil, err := cr.NewContainerUtil()
+	if err != nil {
+		cp.log.Errorf("could not initialise container utility: %v", err)
+		phaseutils.SetHostStatus(cp.hostPhase, constants.FailedState, err.Error())
+		return errors.Wrap(err, "could not initialise container utility")
+	}
+
+	containers, err := containerUtil.GetContainersInNamespace(ctx, constants.K8sNamespace)
+	if err != nil {
+		cp.log.Errorf("error getting containers in namespace: %s :%v", constants.K8sNamespace, err)
+		phaseutils.SetHostStatus(cp.hostPhase, constants.FailedState, err.Error())
+		return errors.Wrapf(err, "error getting containers in namespace: %s ", constants.K8sNamespace)
+	}
+
+	ctx = namespaces.WithNamespace(ctx, constants.K8sNamespace)
+
+	cp.log.Infof("Destroying containers")
+	err = containerUtil.EnsureContainersDestroyed(ctx, containers, timeOut)
+	if err != nil {
+		cp.log.Errorf("could not destroy containers in namespace: %s :%v", constants.K8sNamespace, err)
+		phaseutils.SetHostStatus(cp.hostPhase, constants.FailedState, err.Error())
+		return errors.Wrapf(err, "could not destroy containers in namespace: %s", constants.K8sNamespace)
+	}
+
+	// Stop the containerd service
+	cp.log.Infof("Stopping containerd")
+
+	exitCode, err = cp.cmd.RunCommand(ctx, nil, 0, "", "/bin/sudo", "/usr/bin/systemctl", "stop", constants.RuntimeContainerd)
```

Comment:
To keep things more readable implement 3 functions in this file to start/stop/check status containerd.